webapi/src/main/scala/org/knora/webapi/responders/v1/StandoffResponderV1.scala

```diff
@@ -0,0 +1,634 @@
+
+                        Some(XMLStandoffDataTypeClass(
+                            standoffDataTypeClass = dataType,
+                            dataTypeXMLAttribute = dataTypeAttribute
+                        ))
+
+                    case None => None
+                }
+
+                // add the current tag to the map
+                val newNamespaceMap: Map[String, Map[String, XMLTag]] = namespaceMap.get(tagname) match {
+                    case Some(tagMap: Map[String, XMLTag]) =>
+                        tagMap.get(classname) match {
+                            case Some(existingClassname) => throw BadRequestException("Duplicate tag and classname combination in the same namespace")
+                            case None =>
+                                // create the definition for the current element
+                                val xmlElementDef = XMLTag(name = tagname, mapping = XMLTagToStandoffClass(standoffClassIri = standoffClassIri, attributesToProps = attributes, dataType = dataTypeOption), separatorRequired = curEle.separatorRequired)
+
+                                // combine the definition for the this classname with the existing definitions beloning to the same element
+                                val combinedClassDef: Map[String, XMLTag] = namespaceMap(tagname) + (classname -> xmlElementDef)
+
+                                // combine all elements for this namespace
+                                namespaceMap + (tagname -> combinedClassDef)
+
+                        }
+                    case None =>
+                        namespaceMap + (tagname -> Map(classname -> XMLTag(name = tagname, mapping = XMLTagToStandoffClass(standoffClassIri = standoffClassIri, attributesToProps = attributes, dataType = dataTypeOption), separatorRequired = curEle.separatorRequired)))
+                }
+
+                // recreate the whole structure for all namespaces
+                MappingXMLtoStandoff(
+                    namespace = acc.namespace + (namespace -> newNamespaceMap)
+                )
+
+        }
+
+        // invert mapping in order to run checks for duplicate use of
+        // standoff class Iris and property Iris in the attributes for a standoff class
+        StandoffTagUtilV1.invertXMLToStandoffMapping(mappingXMLToStandoff)
+
+        mappingXMLToStandoff
+
+    }
+
+    /**
+      * The name of the mapping cache.
+      */
+    val MappingCacheName = "mappingCache"
+
+    /**
+      * Gets a mapping either from the cache or by making a requests to the triplestore.
+      *
+      * @param mappingIri  the Iri of the mapping to retrieve.
+      * @param userProfile the user making the request.
+      * @return a [[MappingXMLtoStandoff]].
+      */
+    private def getMappingV1(mappingIri: IRI, userProfile: UserProfileV1): Future[GetMappingResponseV1] = {
+
+        CacheUtil.get[MappingXMLtoStandoff](cacheName = MappingCacheName, key = mappingIri) match {
```

Comment:
Put `CacheUtil.get` inside the `for` comprehension, in case it throws an exception.